lib/input/blob_storage.go

```diff
@@ -0,0 +1,449 @@
+
+By default, the ` + "`timeout`" + ` parameter is set to 0 seconds, which disables timeouts on the Azure Storage side.
+
+## Metadata
+
+This input adds the following metadata fields to each message:
+
+` + "```" + `
+- blob_storage_key
+- blob_storage_container
+- blob_storage_last_modified
+- blob_storage_last_modified_unix
+- blob_storage_content_type
+- blob_storage_content_encoding
+- All user defined metadata
+` + "```" + `
+
+You can access these metadata fields using [function interpolation](/docs/configuration/interpolation#metadata).`,
+		FieldSpecs: docs.FieldSpecs{
+			docs.FieldCommon(
+				"storage_account",
+				"The storage account to download blobs from. This field is ignored if `storage_connection_string` is set.",
+			),
+			docs.FieldCommon(
+				"storage_access_key",
+				"The storage account access key. This field is ignored if `storage_connection_string` is set.",
+			),
+			docs.FieldCommon(
+				"storage_connection_string",
+				"A storage account connection string. This field is required if `storage_account` and `storage_access_key` are not set.",
+			),
+			docs.FieldCommon(
+				"container", "The name of the container from which to download blobs.",
+			),
+			docs.FieldCommon("prefix", "An optional path prefix, if set only objects with the prefix are consumed."),
+			codec.ReaderDocs,
+			docs.FieldAdvanced("delete_objects", "Whether to delete downloaded objects from the blob once they are processed."),
+			docs.FieldAdvanced("timeout", "The maximum period to wait on a download before abandoning it and reattempting."),
+		},
+		Categories: []Category{
+			CategoryServices,
+			CategoryAzure,
+		},
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// AzureBlobStorageConfig contains configuration fields for the AzureBlobStorage
+// input type.
+type AzureBlobStorageConfig struct {
+	StorageAccount          string `json:"storage_account" yaml:"storage_account"`
+	StorageAccessKey        string `json:"storage_access_key" yaml:"storage_access_key"`
+	StorageConnectionString string `json:"storage_connection_string" yaml:"storage_connection_string"`
+	Container               string `json:"container" yaml:"container"`
+	Prefix                  string `json:"prefix" yaml:"prefix"`
+	Codec                   string `json:"codec" yaml:"codec"`
+	DeleteObjects           bool   `json:"delete_objects" yaml:"delete_objects"`
+	Timeout                 string `json:"timeout" yaml:"timeout"`
+}
+
+// NewAzureBlobStorageConfig creates a new AzureBlobStorageConfig with default
+// values.
+func NewAzureBlobStorageConfig() AzureBlobStorageConfig {
+	return AzureBlobStorageConfig{
+		Codec:   "all-bytes",
+		Timeout: "0s",
+	}
+}
+
+//------------------------------------------------------------------------------
+
+type azureObjectTarget struct {
```

Comment:
Should I rename `objectTarget` to `s3ObjectTarget` in [`aws_s3.go`](https://github.com/Jeffail/benthos/blob/2955a506cc48cf2afafdddea4eac4eb8ea8e9d5d/lib/input/aws_s3.go#L168)?

Reply:
Please, that would make things a bit clearer.